tests/relay/schema.py

```diff
@@ -0,0 +1,185 @@
+from typing import Iterable, List, Optional
+from typing_extensions import Self
+
+import strawberry
+from strawberry.relay.utils import to_base64
+from strawberry.types import Info
+
+
+@strawberry.type
+class Fruit(strawberry.relay.Node):
+    _id: strawberry.Private[int]
+    name: str
+    color: str
+
+    @classmethod
+    def resolve_id(cls, root: Self, *, info: Optional[Info] = None):
+        return root._id
+
+    @classmethod
+    def resolve_nodes(
+        cls,
+        *,
+        info: Optional[Info] = None,
+        node_ids: Optional[Iterable[str]] = None,
+    ):
+        if node_ids is not None:
+            return [fruits[nid] for nid in node_ids]
+
+        return list(fruits.values())
+
+    @classmethod
+    def resolve_node(
+        cls,
+        node_id: str,
+        *,
+        info: Optional[Info] = None,
+        required: bool = False,
+    ):
+        obj = fruits.get(node_id, None)
+        if required and obj is None:
+            raise ValueError(f"No fruit by id {node_id}")
+
+        return obj
+
+
+@strawberry.type
+class FruitAsync(Fruit):
+    @classmethod
+    async def resolve_id(cls, root: Self, *, info: Optional[Info] = None):
+        return super().resolve_id(root, info=info)
+
+    @classmethod
+    async def resolve_nodes(
+        cls,
+        *,
+        info: Optional[Info] = None,
+        node_ids: Optional[Iterable[str]] = None,
+    ):
+        if node_ids is not None:
+            return [fruits_async[nid] for nid in node_ids]
+
+        return list(fruits_async.values())
+
+    @classmethod
+    async def resolve_node(
+        cls,
+        node_id: str,
+        *,
+        info: Optional[Info] = None,
+        required: bool = False,
+    ):
+        obj = fruits_async.get(node_id, None)
+        if required and obj is None:
+            raise ValueError(f"No fruit by id {node_id}")
+
+        return obj
+
+
+@strawberry.type
+class CustomPaginationConnection(
+    strawberry.relay.Connection[strawberry.relay.NodeType]
+):
+    @strawberry.field
+    def something(self) -> str:
+        return "foobar"
+
+    @classmethod
+    def from_nodes(
+        cls,
+        nodes: Iterable[Fruit],
```

Comment:
This should be `Iterable[NodeType]` given that the type is not specialized here, but still a `Generic[NodeType]`, which is wrong as we want it to be specialized now.

I think the correct use here is:
```python
@strawberry.type
class CustomPaginationConnection(
    strawberry.relay.Connection[Fruit]
):
    ...


@strawberry.type
class Query:
    ...
    fruits_custom_pagination: CustomPaginationConnection

```

This would explain my confusion I had with this, and also it would then make sense to perhaps call it `CustomFruitPaginationConnection` ;)

Reply:
Yeah, I agree. I actually tried to do that when I originally wrote this test on `strawberry-django-plus`, but strawberry was not able to generate the type that way.

Maybe now that this code is being merged here I can try to modify the Generic handling to support that code.

Reply:
I investigated and discovered the issue here. It was not specific to the relay implementation, but with `Generic` type trying to create a specialized type in that way.

Basically, when creating the fields, the `type_var_map` was not being set for them. I fixed the issue and added tests for it in `test_generics`: https://github.com/strawberry-graphql/strawberry/pull/2511/commits/6bc2f4f0b4f311125144c64f1aa07cff8ec9fa06